recipes/openssl/3.x.x/conanfile.py

```diff
@@ -0,0 +1,639 @@
+            if not self.options.no_asm and not tools.which("nasm"):
+                self.build_requires("nasm/2.15.05")
+        if self._win_bash:
+            if not tools.get_env("CONAN_BASH_PATH"):
+                self.build_requires("msys2/cci.latest")
+
+    def requirements(self):
+        if self.options.zlib:
+            self.requires("zlib/1.2.11")
+
+    @property
+    def _is_msvc(self):
+        return self.settings.compiler == "Visual Studio"
+
+    @property
+    def _is_clangcl(self):
+        return self.settings.compiler == "clang" and self.settings.os == "Windows"
+
+    @property
+    def _is_mingw(self):
+        return self.settings.os == "Windows" and self.settings.compiler == "gcc"
+
+    @property
+    def _use_nmake(self):
+        return self._is_clangcl or self._is_msvc
+
+    def config_options(self):
+        if self.settings.os != "Windows":
+            del self.options.capieng_dialog
+            del self.options.enable_capieng
+        else:
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    @property
+    def _target(self):
+        target = "conan-%s-%s-%s-%s-%s" % (self.settings.build_type,
+                                           self.settings.os,
+                                           self.settings.arch,
+                                           self.settings.compiler,
+                                           self.settings.compiler.version)
+        if self._use_nmake:
+            target = "VC-" + target  # VC- prefix is important as it's checked by Configure
+        if self._is_mingw:
+            target = "mingw-" + target
+        return target
+
+    @property
+    def _perlasm_scheme(self):
+        # right now, we need to tweak this for iOS & Android only, as they inherit from generic targets
+        the_arch = str(self.settings.arch)
+        the_os = str(self.settings.os)
+        if the_os in ["iOS", "watchOS", "tvOS"]:
+            return {"armv7": "ios32",
+                    "armv7s": "ios32",
+                    "armv8": "ios64",
+                    "armv8_32": "ios64",
+                    "armv8.3": "ios64",
+                    "armv7k": "ios32"}.get(the_arch, None)
+        elif the_os == "Android":
+            return {"armv7": "void",
+                    "armv8": "linux64",
+                    "mips": "o32",
+                    "mips64": "64",
+                    "x86": "android",
+                    "x86_64": "elf"}.get(the_arch, None)
+        return None
+
+    @property
+    def _asm_target(self):
+        the_os = str(self.settings.os)
+        if the_os in ["Android", "iOS", "watchOS", "tvOS"]:
```

Comment:
The strinigication is not needed, I think.
Bonus is extra conan checks.
```suggestion
        if self.settings.os in ("Android", "iOS", "watchOS", "tvOS"):
```